### PR TITLE
Airtable Integration

### PR DESCRIPTION
**Pull request by TalaatHasanin**
# Description
Airtable Data Integration Source

# How Has This Been Tested?

- [x] Test Integration

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

cc:
@wangxiaoyou1993

### CODE REVIEW

**Comment by TalaatHasanin** [comment]
I tested the integration using UI in the following screens:
Screenshot 2024-09-13 145924
Screenshot 2024-09-13 145940
Screenshot 2024-09-13 150008

**Comment by TalaatHasanin** [comment]
I have an error when trying to load sample data for the last stream in the list which is Sales Activities. Can You review it? @wangxiaoyou1993 

The error:
Screenshot 2024-09-13 150110

the full error message:
```
Error

The sample data was not able to be loaded. Please check if the stream still exists. If it does not, click the "View and select streams" button and confirm the valid streams. If it does, loading sample data for this source may not currently be supported.

Traceback ([hide traceback](http://localhost:3000/pipelines/relaxed_grimoire/edit?sideview=data#))

Traceback (most recent call last):

  File "/home/src/mage_ai/data_preparation/models/pipelines/integration_pipeline.py", line 260, in preview_data

    proc.check_returncode()

  File "/usr/local/lib/python3.10/subprocess.py", line 457, in check_returncode

    raise CalledProcessError(self.returncode, self.args, self.stdout,

subprocess.CalledProcessError: Command '['python3', '/usr/local/lib/python3.10/site-packages/mage_integrations/sources/airtable/__init__.py', '--config_json', '{"token": "patl1U8AGjswX9GZM.694b98419313ffa59f186466705c4908abdf2ac708f42685dfcf2c2bc63e9047", "base_id": "appVPmW2WPnjcGMTd", "table_name": null}', '--load_sample_data', '--log_to_stdout', '1', '--settings', '/home/src/default_repo/pipelines/relaxed_grimoire/data_integration_catalog.json', '--selected_streams_json', '["Sales_Activities"]']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/usr/local/lib/python3.10/site-packages/tornado/web.py", line 1786, in _execute

    result = await result

  File "/home/src/mage_ai/server/api/v1.py", line 64, in post

    return await execute_operation(self, resource)

  File "/home/src/mage_ai/api/views.py", line 70, in execute_operation

    raise err

  File "/home/src/mage_ai/api/views.py", line 63, in execute_operation

    response = await base_operation.execute()

  File "/home/src/mage_ai/api/operations/base.py", line 93, in execute

    result = await self.__executed_result()

  File "/home/src/mage_ai/api/operations/base.py", line 481, in __executed_result

    return await self.__create_or_index()

  File "/home/src/mage_ai/api/operations/base.py", line 534, in __create_or_index

    result = await self.__resource_class().process_create(

  File "/home/src/mage_ai/api/resources/BaseResource.py", line 212, in process_create

    raise err

  File "/home/src/mage_ai/api/resources/BaseResource.py", line 189, in process_create

    res = cls.create(

  File "/home/src/mage_ai/orchestration/db/__init__.py", line 158, in func_with_rollback

    return func(*args, **kwargs)

  File "/home/src/mage_ai/api/resources/IntegrationSourceResource.py", line 173, in create

    streams_updated = pipeline.preview_data(

  File "/home/src/mage_ai/data_preparation/models/pipelines/integration_pipeline.py", line 298, in preview_data

    raise Exception('The sample data was not able to be loaded. Please check if the ' +

Exception: The sample data was not able to be loaded. Please check if the stream still exists. If it does not, click the "View and select streams" button and confirm the valid streams. If it does, loading sample data for this source may not currently be supported.

Stack trace ([hide stack trace](http://localhost:3000/pipelines/relaxed_grimoire/edit?sideview=data#))

  File "/home/src/mage_ai/server/server.py", line 833, in      start_server(

  File "/home/src/mage_ai/server/server.py", line 801, in start_server     asyncio.run(

  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run     return loop.run_until_complete(main)

  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 636, in run_until_complete     self.run_forever()

  File "/usr/local/lib/python3.10/asyncio/base_events.py", line 603, in run_forever     self._run_once()

  File "/usr/local/lib/python3.10/site-packages/nest_asyncio.py", line 133, in _run_once     handle._run()

  File "/usr/local/lib/python3.10/asyncio/events.py", line 80, in _run     self._context.run(self._callback, *self._args)

  File "/home/src/mage_ai/server/server.py", line 720, in main     get_messages(

  File "/home/src/mage_ai/server/subscriber.py", line 14, in get_messages     message = client.get_iopub_msg(timeout=1)

  File "/usr/local/lib/python3.10/site-packages/jupyter_client/utils.py", line 27, in wrapped     return loop.run_until_complete(future)

  File "/usr/local/lib/python3.10/site-packages/nest_asyncio.py", line 92, in run_until_complete     self._run_once()

  File "/usr/local/lib/python3.10/site-packages/nest_asyncio.py", line 133, in _run_once     handle._run()

  File "/usr/local/lib/python3.10/asyncio/events.py", line 80, in _run     self._context.run(self._callback, *self._args)

  File "/usr/local/lib/python3.10/asyncio/tasks.py", line 232, in __step     result = coro.send(None)

  File "/usr/local/lib/python3.10/site-packages/tornado/web.py", line 1791, in _execute     self._handle_request_exception(e)

  File "/usr/local/lib/python3.10/site-packages/tornado/web.py", line 1848, in _handle_request_exception     self.send_error(500, exc_info=sys.exc_info())

  File "/usr/local/lib/python3.10/site-packages/tornado/web.py", line 1294, in send_error     self.write_error(status_code, **kwargs)

  File "/home/src/mage_ai/server/api/base.py", line 65, in write_error     errors = traceback.format_stack()
```

**Comment by TalaatHasanin** [comment]
completed pipeline:
Screenshot 2024-09-17 111823
Screenshot 2024-09-17 111726

File: mage_integrations/mage_integrations/connections/airtable/__init__.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from pyairtable import Api, Base
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/connections/airtable/__init__.py:1]: do you need to add this package to requirements.txt?

File: mage_integrations/mage_integrations/sources/airtable/__init__.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+from mage_integrations.sources.constants import (
+    COLUMN_TYPE_ARRAY,
+    COLUMN_TYPE_OBJECT,
+    COLUMN_TYPE_STRING,
+    REPLICATION_METHOD_FULL_TABLE,
+    UNIQUE_CONFLICT_METHOD_UPDATE,
+)
+from mage_integrations.sources.utils import get_standard_metadata
+
+LOGGER = singer.get_logger()
```
Comment (wangxiaoyou1993) [inline mage_integrations/mage_integrations/sources/airtable/__init__.py:19]: nit: this is not used. if you want to use logger, you can use self.logger in your code, which inherits from the base source.

File: mage_integrations/mage_integrations/connections/airtable/__init__.py (hunk 1 above)
Comment (TalaatHasanin) [reply mage_integrations/mage_integrations/connections/airtable/__init__.py:1]: yes

**Review by wangxiaoyou1993 — APPROVED** [review]

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**